tests/src/test/java/alluxio/master/PortRegistryTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Unit tests for {@link PortRegistry}.
+ */
+public final class PortRegistryTest {
+  private PortRegistry mRegistry = new PortRegistry();
```

Comment:
Why make the port registry an instantiable object when the locking occurs on a system-wide basis? Wouldn't it make more sense to make the methods on the registry static?

Reply:
it has state so it needs to be instantiated. We could add convenience methods similar to ServerConfiguration, but it requires duplication of every method signature

Reply:
I think I'd prefer using the static methods. I can live without them though